### PR TITLE
msg: remove stale comment and unnecessary code

### PR DESCRIPTION
**Pull request by changchengx**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/msg/Message.h [hunk 1]
```diff
@@ -546,7 +546,7 @@ class SafeMessage : public Message {
 
 namespace ceph {
 template<class T, typename... Args>
-ceph::ref_t<T> make_message(Args&&... args) {
+MRef<T> make_message(Args&&... args) {
```
Comment (tchaikov) [inline src/msg/Message.h:547]: i'd prefer using `ceph::ref_t`. there is not need to have two groups of aliases mirroring each other in the same source tree just for emphasizing that one is dedicated to `Message` while another is for all types in general.

File: src/msg/Message.h [hunk 2]
```diff
@@ -318,8 +318,6 @@ class Message : public RefCountedObject {
     header.type = t;
     header.version = version;
     header.compat_version = compat_version;
-    header.priority = 0;  // undef
```
Comment (tchaikov) [inline src/msg/Message.h:321]: ack.

File: src/msg/Message.h (hunk 1 above)
Comment (changchengx) [reply src/msg/Message.h:547]: @tchaikov I'll abandon this commit.

File: src/msg/Messenger.h [hunk 3]
```diff
@@ -156,7 +156,6 @@ class Messenger {
    * @param name entity name to register
    * @param lname logical name of the messenger in this process (e.g., "client")
    * @param nonce nonce value to uniquely identify this instance on the current host
-   * @param features bits for the local connection
```
[the remark this answers is not in the document]
Comment (changchengx) [reply src/msg/Messenger.h:159]: Thanks. I'll apply your suggested comment

**Review by tchaikov — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2020-10-13_01:10:32-rados-wip-kefu-testing-2020-10-12-1615-distro-basic-smithi/

**Pull request merged**